src/sagemaker/jumpstart/curated_hub/utils.py

```diff
@@ -164,3 +179,97 @@ def create_hub_bucket_if_it_does_not_exist(
     )
 
     return bucket_name
+
+def tag_hub_content(hub_content_arn: str, tags: List[Tag], session: Session) -> List[Dict[str, List[Dict[str, str]]]]:
+    responses = []          
+    for tag in tags:
+      responses.add(session.add_tags(
+        ResourceArn=hub_content_arn,
+        Tags=[
+            {
+                'Key': tag.key,
+                'Value': tag.value
+            },
+        ]
+      ))
+
+    return responses
+    
+def find_jumpstart_tags_for_model(hub_name: str, hub_content_name: str, region: str, session: Session) -> List[Tag]:
+    list_versions_response = session.list_hub_content_versions(
+        hub_name=hub_name,
+        hub_content_type='Model',
+        hub_content_name=hub_content_name
+    )
+    hub_content_versions = list_versions_response["HubContentSummaries"]
+
+    tag_name_to_versions_map: Dict[CuratedHubTagName, List[str]] = {}
+    for hub_content_version_summary in hub_content_versions:
+        jumpstart_model = get_jumpstart_model_and_version(hub_content_version_summary)
+        if jumpstart_model["model_id"] is None or jumpstart_model["version"] is None:
+            continue
+        tag_names_to_add: List[CuratedHubTagName] = find_jumpstart_tags_for_model_version(
+            model_id=jumpstart_model["model_id"],
+            version=jumpstart_model["version"],
+            region=region,
+            session=session
+        )
+
+        for tag_name in tag_names_to_add:
+          if tag_name not in tag_name_to_versions_map:
+              tag_name_to_versions_map[tag_name] = []
+          tag_name_to_versions_map[tag_name].append(hub_content_version_summary["HubContentVersion"])
+      
+    tags: List[Tag] = []
+    for tag_name, versions in tag_name_to_versions_map.items():
+        tags.append(Tag(
+            key=tag_name,
+            value=str(versions)
+        ))
+
+    return tags
+
+    
+    
+def find_jumpstart_tags_for_model_version(model_id: str, version: str, region: str, session: Session) -> List[CuratedHubTagName]:
+    tags_to_add: List[CuratedHubTagName] = []
+    specs = utils.verify_model_region_and_return_specs(
+        model_id=model_id, 
+        version=version,
+        region=region,
+        scope=JumpStartScriptScope.INFERENCE,
```

Comment:
Question: will this fail to find any training vulnerabilities?

Reply:
Good question. This will pull the specs for the document, which we will then check below